### ISSUE TITLE
Tests for .sol files

### ISSUE THREAD

**Issue by marcelomorgado** [studydefi/money-legos#48]
Hi there, I've just pushed the Kyber interface fix. I've just noticed that was wrong because I've tried to use if for a project of mine.
I think that it would be better if contract errors could be caught by a test script. Do you think that make sense to have some truffle test cases for solidity files to have they covered?

Best,

**Comment by adrianmcli** [comment]
That's a good idea to have Solidity test, I've been meaning to do that for a while. However, I would like to avoid Truffle's test runner if possible (especially with the way the repo is currently organized). I wonder if we can kind of roll our own testing harness.

**Comment by marcelomorgado** [comment]
Hi @adrianmcli, I've created this PoC here: https://github.com/studydefi/money-legos/pull/49 to have the solidity cover as well.
Some comments:
- I didn't found a straightforward buidler-jest integration so I had to change to mocha;
- Since Buidler EVM doesn't support fork from mainnet, I've setup ganache integration for that purpose (because of that console.log isn't available).

WDYT?

**Comment by adrianmcli** [comment]
@marcelomorgado thanks for making this.

It looks like you're still running the tests from Javascript, so I was wondering why you needed to use Buidler and Waffle. They are great projects, but I'm not sure it's doing more than it was doing before since our tests already use Ganache.

**Comment by marcelomorgado** [comment]
Yes, indeed it isn't changing too much, the reasons that I've done that were:
- Buidler is flexible and supports the project's folders structure;
- Better `ethers.js` integration;
- Seems that will DRY things a little bit (i.e. `test-environment.js` will be no longer needed);
- A truffle alternative;

**Comment by adrianmcli** [comment]
My concern is that the tests will take much longer since Mocha and the Ganache-buidler plugin does not run in parallel.

Currently, Jest allows us to create a Ganache instance for each test suite (i.e. test file) and run them in parallel. Also we don't use Truffle here either because this is not a dapp project. If I have a choice, I would always choose Buidler over Truffle.

**Comment by marcelomorgado** [comment]
I see, lose the parallel tests would be a regression here. So, the only thing that builder could help is to compile solidity code to see if there is an obvious error (like the one that I've found in the Kyber doc page sol code example that I've used in the PoC) but I'm not sure if worths.
I'll close my MR though (and open another one with the doc fix).
Thanks for the feedback @adrianmcli

**Comment by adrianmcli** [comment]
@marcelomorgado no problem, thanks for making this, sorry we can't merge it in.

I'm actually working with Pato from Buidler, and maybe we can have something by next month that can actually help with this. I understand we need Buidler to compile Solidity contracts if we want to test Solidity usage. I'll keep this in mind, so we will leave this issue open.

**Comment by marcelomorgado** [comment]
Hi @adrianmcli !
WDYT to use `hardhat` to cover the `.sol` files and have faster tests?

**Comment by adrianmcli** [comment]
@marcelomorgado I think that's a great idea! Hardhat forking is so much faster than Ganache.
